tests/testthat/test-pstr_add_transition_risk.R

```diff
@@ -8,3 +8,60 @@ test_that("hasn't change", {
   out <- format_robust_snapshot(with_transition_risk)
   expect_snapshot(out)
 })
+
+test_that("returns a tibble data frame", {
+  out <- pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo(), pstr_toy_weo_2022())
+  expect_s3_class(
+    pstr_add_transition_risk(out),
+    "tbl_df"
+  )
+})
+
+test_that("adds one column: `transition_risk`", {
+  with_reductions <- pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo(), pstr_toy_weo_2022())
+  expect_false(hasName(with_reductions, "transition_risk"))
+
+  out <- pstr_add_transition_risk(with_reductions)
+  new_columns <- setdiff(names(out), names(with_reductions))
+  expect_equal(
+    new_columns, "transition_risk"
+  )
+})
+
+test_that("added column returns acceptable values", {
+  out <- pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo(), pstr_toy_weo_2022())
+  acceptable_transition_risks <- c(
+    "low",
+    "medium",
+    "high",
+    "no_sector"
+  )
+
+  expect_true(
+    all(
+      pstr_add_transition_risk(out)$transition_risk %in% acceptable_transition_risks
+    )
+  )
+})
+
+test_that("preserves typeof() input columns", {
+  with_reductions <- pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo(), pstr_toy_weo_2022())
+  out <- pstr_add_transition_risk(with_reductions)
+
+  x <- unname(purrr::map_chr(with_reductions, typeof))
+  y <- unname(purrr::map_chr(out[names(with_reductions)], typeof))
```

Comment:
Nit: You may import `map_chr()` with `usethis::use_import_from("purrr", "map_chr")`. 
If you use it a few times it's worth importing.